src/components/Settings/Settings.tsx

```diff
@@ -171,113 +201,151 @@ function SettingsContent({
         }
     });
 
-    const renderPageContent = () => {
-        if (!activePage) {
+    React.useEffect(() => {
+        if (!selectionInitialPage) return;
+        setCurrentPage(selectionInitialPage);
+    }, [selectionInitialPage]);
+
+    React.useEffect(() => {
+        if (!selected.selectedRef?.current) return;
```

Comment:
It's better to invert the condition here

Reply:
But that way we get an unnecessary indentation

Reply:
We don't skimp on indentations. The code without negation in the condition in this case is more idomatic and easier to understand

Reply:
Inverted the condition